cni-plugin/tests/calico_cni_k8s_test.go

```diff
@@ -2999,6 +3001,130 @@ var _ = Describe("Kubernetes CNI tests", func() {
 		})
 	})
 
+	Context("using bogus readiness_gates", func() {
+		netconf := fmt.Sprintf(`
+				{
+				  "cniVersion": "%s",
+				  "name": "net10",
+				  "type": "calico",
+				  "etcd_endpoints": "http://%s:2379",
+				  "datastore_type": "%s",
+           			  "nodename_file_optional": true,
+				  "log_level": "info",
+				  "readiness_gates": "http://localhost:9099/invalid_x12vx",
+			 	  "ipam": {
+				    "type": "calico-ipam"
+				  },
+				  "kubernetes": {
+				    "kubeconfig": "/home/user/certs/kubeconfig"
+				  },
+				  "policy": {"type": "k8s"}
+				}`, cniVersion, os.Getenv("ETCD_IP"), os.Getenv("DATASTORE_TYPE"))
+
+		It("should fail container creation", func() {
+			// Create a new ipPool.
+			testutils.MustCreateNewIPPool(calicoClient, "10.0.0.0/24", false, false, true)
+
+			clientset := getKubernetesClient()
+
+			// Now create a K8s pod.
+			name := "mypod-1"
+
+			ensureNamespace(clientset, testutils.K8S_TEST_NS)
+			ensurePodCreated(clientset, testutils.K8S_TEST_NS,
+				&v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: name,
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{{
+							Name:  name,
+							Image: "ignore",
+						}},
+						NodeName: hostname,
+					},
+				})
+
+			// Create the container, which will call CNI and by default it will create the container with interface name 'eth0'.
+			containerID, _, _, _, _, contNs, err := testutils.CreateContainer(netconf, name, testutils.K8S_TEST_NS, "")
+			Expect(err).Should(HaveOccurred())
+			// Make sure the pod gets cleaned up, whether we fail or not.
+			expectedIfaceName := "eth0"
+			defer func() {
```

Comment:
This doesn't need to be in a defer statement, because we don't do any work after this.